tests/pass-dep/libc/libc-mem.rs

```diff
@@ -241,6 +241,42 @@ fn test_reallocarray() {
     }
 }
 
+#[cfg(not(any(target_os = "windows", target_os = "wasi")))]
+fn test_aligned_alloc() {
+    // libc doesn't have this function (https://github.com/rust-lang/libc/issues/3689),
+    // so we declare it ourselves.
+    extern "C" {
+        fn aligned_alloc(alignment: libc::size_t, size: libc::size_t) -> *mut libc::c_void;
+    }
+    // size not a multiple of the alignment
+    unsafe {
+        let p = aligned_alloc(16, 3);
+        assert_eq!(p, ptr::null_mut());
+    }
+
+    // alignment not power of 2
+    unsafe {
+        let p = aligned_alloc(63, 8);
+        assert_eq!(p, ptr::null_mut());
+    }
+
+    // alignment lesser than a word but still a successful allocation
+    unsafe {
+        let p = aligned_alloc(1, 4);
+        assert!(!p.is_null());
+        assert!(p.is_aligned_to(4));
+        libc::free(p);
+    }
+
+    // finally ..
+    unsafe {
+        let p = aligned_alloc(16, 16);
+        assert!(!p.is_null());
+        assert!(p.is_aligned_to(16));
+        libc::free(p);
+    }
```

Comment:
Maybe put these in a `for _ in 0..16` or so, because the actual alignment is random so we want to get some extra test coverage.